pubsub_nats.go

```diff
@@ -138,15 +147,26 @@ func (p *natsPubSub) Subscribe(pubs chan *Publication, errors chan error, topic
 		}
 
 		if m.Reply != "" {
-
-			resp := ackMessage
-			if config.replier != nil {
-				resp = config.replier(m)
-			}
-
-			if err := p.client.Publish(m.Reply, resp); err != nil {
-				zap.L().Error("Unable to send requested reply", zap.Error(err))
-				return
+			switch publication.ResponseMode {
+			// `ResponseModeACK` mode responds to the client right away, BEFORE the subscriber has had the opportunity
+			// to process the publication. Most consumers do message processing asynchronously and simply need
+			// to respond to the publisher with an ACK.
+			case ResponseModeACK:
+				if err := p.client.Publish(m.Reply, ackMessage); err != nil {
+					zap.L().Error("Failed to publish ACK reply. Message dropped.",
+						zap.Error(err),
+						zap.String("nats_subject", m.Reply))
```

Comment:
@primalmotion shouldn't we also be sending an error to the errors channel? Presently we don't do this, that's why I didn't add it here. But, I think we should as this seems to have been an oversight in the original implementation.

Reply:
Done!